### PR TITLE
Recording feature

### PR DESCRIPTION
**Pull request by jowczar**

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/drivers/DriverManager.java [hunk 1]
```diff
@@ -10,12 +11,14 @@
 public class DriverManager {
 
 	private Job2dDriver currentDriver = new LoggerDriver();
+	private final Publisher changePublisher = new Publisher();
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/DriverManager.java:14]: Another group is working on "Observer for a driver changes events" so I want to give them a chance to do this task. Therefore, I will wait with merging your solution.

File: src/main/java/edu/kis/powp/jobs2d/drivers/DriverChangeObserver.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+public class DriverChangeObserver implements Subscriber {
+
+    @Override
+    public void update() {
+        RecordingManager recordingManager = RecordingFeature.getRecordingManager();
+
+        Job2dDriver driver = DriverFeature.getDriverManager().getCurrentDriver();
+        if (recordingManager.getStatus() == RECORDING_STATUS.IN_PROGRESS
+                && !driver.getClass().getSimpleName().equals("CommandsRecorderDriver")) {
+            recordingManager.resumeRecording();
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/DriverChangeObserver.java]: A notification about the recording process and the necessity to substitute a driver would be helpful, eg. "Recording command in progress. To use the driver stop recording."
You can use JOptionPane.

**Review by kowallus — COMMENTED** [review]
I will wait with merging for "Observer for a driver changes events" task. You will probably need to make minor changes (conflicts, etc.) then.

File: src/main/java/edu/kis/powp/jobs2d/drivers/DriverChangeObserver.java (hunk 2 above)
Comment (jowczar) [reply src/main/java/edu/kis/powp/jobs2d/drivers/DriverChangeObserver.java]: @kowallus  I think now that we use DriverComposite and recording runs in parallel with drawing, this notification becomes obsolete. Is this right?

File: src/main/java/edu/kis/powp/jobs2d/drivers/DriverChangeObserver.java [hunk 3]
```diff
@@ -1,25 +1,21 @@
 package edu.kis.powp.jobs2d.drivers;
 
-import edu.kis.powp.jobs2d.Job2dDriver;
 import edu.kis.powp.jobs2d.enums.RECORDING_STATUS;
-import edu.kis.powp.jobs2d.features.DriverFeature;
 import edu.kis.powp.jobs2d.features.RecordingFeature;
 import edu.kis.powp.observer.Subscriber;
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/DriverChangeObserver.java]: You should find a better name for DriverChangeObserver (such a name suits literally every driver change observer). It doesn't tell anything about its purpose.

**Review by kowallus — CHANGES_REQUESTED** [review]
Nice. I see only one important issue to fix.

**Pull request merged**